notify/webhook/webhook.go

```diff
@@ -53,8 +54,8 @@ func New(conf *config.WebhookConfig, t *template.Template, l log.Logger, httpOpt
 		// Webhooks are assumed to respond with 2xx response codes on a successful
 		// request and 5xx response codes are assumed to be recoverable.
 		retrier: &notify.Retrier{
-			CustomDetailsFunc: func(int, io.Reader) string {
-				return conf.URL.String()
+			CustomDetailsFunc: func(_ int, body io.Reader) string {
+				return errDetails(body, conf.URL.String())
```

Comment:
It might be worth passing the status code to `errDetails` so it can be incorporated in the error message?

Reply:
The message returned by `errDetails` is concatenated with the status code afterwards already:
https://github.com/prometheus/alertmanager/blob/1045dc0f2144e1463941d56210e0e6ebaff2a73e/notify/util.go#L200-L209